### PR TITLE
ENT-6542: Add Community Edition string for R3 builds

### PR DESCRIPTION
**Pull request by fowlerrr**
Not quite ready yet, but raising the PR might be the best way to test it

### CODE REVIEW

File: .ci/dev/regression/Jenkinsfile [hunk 1]
```diff
@@ -10,6 +10,13 @@
 boolean isReleaseTag = (env.TAG_NAME =~ /^release-.*(?<!_JDK11)$/)
 boolean isInternalRelease = (env.TAG_NAME =~ /^internal-release-.*$/)
 boolean isReleaseCandidate = (env.TAG_NAME =~ /^(release-.*(RC|HC).*(?<!_JDK11))$/)
+String getBuildEdition() {
+    return if (isReleaseTag || isReleaseCandidate) {
+        "Corda Community Edition, built by R3"
+    } else {
+        "Corda Open Sauce"
```
Comment (fowlerrr) [inline .ci/dev/regression/Jenkinsfile]: This needs changing.  It's meant to be different so I can see if we ever reach the else condition

File: .ci/dev/regression/Jenkinsfile [hunk 2]
```diff
@@ -10,6 +10,13 @@
 boolean isReleaseTag = (env.TAG_NAME =~ /^release-.*(?<!_JDK11)$/)
 boolean isInternalRelease = (env.TAG_NAME =~ /^internal-release-.*$/)
 boolean isReleaseCandidate = (env.TAG_NAME =~ /^(release-.*(RC|HC).*(?<!_JDK11))$/)
+String getBuildEdition() {
```
Comment (ronanbrowne) [inline .ci/dev/regression/Jenkinsfile]: def getBuildEdition =  (isReleaseTag || isReleaseCandidate) ? 'Corda Community Edition, built by R3' : "Corda Open Source"  


Above should do it, can just use, following if dont want "Corda Open Source" 

def getBuildEdition =  (isReleaseTag || isReleaseCandidate) ? 'Corda Community Edition, built by R3' : ''

File: .ci/dev/regression/Jenkinsfile [hunk 3]
```diff
@@ -76,6 +83,7 @@ pipeline {
         ARTIFACTORY_CREDENTIALS = credentials('artifactory-credentials')
         CORDA_ARTIFACTORY_PASSWORD = "${env.ARTIFACTORY_CREDENTIALS_PSW}"
         CORDA_ARTIFACTORY_USERNAME = "${env.ARTIFACTORY_CREDENTIALS_USR}"
+        CORDA_BUILD_EDITION = getBuildEdition()
```
Comment (ronanbrowne) [inline .ci/dev/regression/Jenkinsfile]: should be   CORDA_BUILD_EDITION = "${getBuildEdition}"

for groovy to pick it up correctly, jenkins failures with out this

File: .ci/dev/regression/Jenkinsfile [hunk 4]
```diff
@@ -10,6 +10,7 @@
 boolean isReleaseTag = (env.TAG_NAME =~ /^release-.*(?<!_JDK11)$/)
 boolean isInternalRelease = (env.TAG_NAME =~ /^internal-release-.*$/)
 boolean isReleaseCandidate = (env.TAG_NAME =~ /^(release-.*(RC|HC).*(?<!_JDK11))$/)
+def buildEdition = (isReleaseTag || isReleaseCandidate) ? "Corda Community Edition, built by R3" : "Corda Open Source"
```
Comment (rick-r3) [inline .ci/dev/regression/Jenkinsfile]: I think just "Corda Community Edition" without the "built by R3" is fine, unless someone else said it should be like that (I know, probably my fault for saying something along those lines).

**Review by rick-r3 — COMMENTED** [review]
Looks like still a work in progress with some stuff commented out in the build that needs cleaning up?

But looks like basically the right thing

File: .ci/dev/regression/Jenkinsfile (hunk 4 above)
Comment (fowlerrr) [reply .ci/dev/regression/Jenkinsfile]: It was in the ticket.  Though not specifically asking for it.  Just an e.g.

**Comment by fowlerrr** [comment]
> Looks like still a work in progress with some stuff commented out in the build that needs cleaning up?
> 
> But looks like basically the right thing

Yeah, that stuff is just there while I'm testing.  So we don't accidentally deploy somewhere.  Just need to run up a published jar to make sure we get the string we expect.  Then I'll tidy up the rest (and change the string, if you want)

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**